x-pack/metricbeat/mb/lightmodules.go

```diff
@@ -0,0 +1,224 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package mb
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+const (
+	moduleYML   = "module.yml"
+	manifestYML = "manifest.yml"
+)
+
+// LightModulesSource loads module definitions from files in the provided paths
+type LightModulesSource struct {
+	paths []string
+}
+
+// NewLightModulesSource creates a new LightModulesSource
+func NewLightModulesSource(paths ...string) *LightModulesSource {
+	return &LightModulesSource{
+		paths: paths,
+	}
+}
+
+// Modules lists the light modules available on the configured paths
+func (r *LightModulesSource) Modules() ([]string, error) {
+	return r.moduleNames()
+}
+
+// HasModule checks if there is a light module with the given name
+func (r *LightModulesSource) HasModule(moduleName string) bool {
+	names, err := r.moduleNames()
+	if err != nil {
+		logp.Error(errors.Wrap(err, "failed to get list of light module names"))
+		return false
+	}
+	for _, name := range names {
+		if name == moduleName {
+			return true
+		}
+	}
+	return false
+}
+
+// DefaultMetricSets list the default metricsets for a given module
+func (r *LightModulesSource) DefaultMetricSets(moduleName string) ([]string, error) {
+	module, err := r.loadModule(moduleName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get default metricsets for module '%s'", moduleName)
+	}
+	var metricsets []string
+	for _, ms := range module.MetricSets {
+		if ms.Default {
+			metricsets = append(metricsets, ms.Name)
```

Comment:
I understand all metricsets are default in a lightweight module, is that correct? I'm :+1: with that until we find a case where that doesn't make sense, probably we won't :)

Reply:
They are not, `default: true` has to be set in the manifest. There are probably cases in `aws` where we don't want to enable everything by default.

Reply:
I'm blind :see_no_evil: :+1: